src/user-io.c

```diff
@@ -50,92 +51,38 @@ void printf_putstring( char *s )
     printf_buffer(s, strlen(s));
 }
 
-static void flush_graphics()
-{
+static void draw_set_buffer(int t, int a0, int a1, int a2, int a3) {
+    struct gfx_command c = {t, {a0, a1, a2, a3}};
+    graphics_buffer[graphics_buffer_index++] = c;
```

Comment:
I see you have each command as a fixed size, which is a good place to start.
Down the road, we may want to allow for the commands to have variable size,
and just advance the pointer by the size of the specific command.
(This would enable passing strings by value.)